tools/RepositoryManagement/DevHome.RepositoryManagement/Services/RepositoryManagementDataAccessService.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using DevHome.Common.Extensions;
+using DevHome.Database;
+using DevHome.Database.DatabaseModels.RepositoryManagement;
+using DevHome.RepositoryManagement.ViewModels;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.Extensions.Hosting;
+using Serilog;
+
+namespace DevHome.RepositoryManagement.Services;
+
+public class RepositoryManagementDataAccessService
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(RepositoryManagementDataAccessService));
+
+    private readonly IHost _host;
+
+    public RepositoryManagementDataAccessService(IHost host)
+    {
+        // Store the host to make dbContext and RepositoryManagementItemViewModel
+        // dbContext can not be passed in because RepositoryManagementDataAccessService is singleton
+        // and dbContext needs to be scoped.
+        // RepositoryManagementItemViewModel can not be passed in because multiple are made.
+        // The best solution is to make factories for DevHomeContext and RepositoryManagementItemViewModel.
+        // Might be in a future change.
+        _host = host;
+    }
+
+    public void AddRepository(string repositoryName, string cloneLocation)
+    {
+        // Check if repositoryName and cloneLocation is null or empty and
+        // return a correct value indicating such.
+        Repository newRepo = new();
+        newRepo.RepositoryName = repositoryName;
+        newRepo.RepositoryClonePath = cloneLocation;
+
+        RepositoryMetadata newMetadata = new();
+        newMetadata.Repository = newRepo;
+        newMetadata.RepositoryId = newRepo.RepositoryId;
+        newMetadata.IsHiddenFromPage = false;
+
+        newRepo.RepositoryMetadata = newMetadata;
+
+        var dbContext = _host.GetService<DevHomeDatabaseContext>();
+        dbContext.Add(newRepo);
+        dbContext.Add(newMetadata);
+        dbContext.SaveChanges();
+    }
+
+    public List<RepositoryManagementItemViewModel> GetRepositories(bool removeRepositoriesNotInTheirSavedLocation)
+    {
+        // This class should not know about RepositoryManagementItemViewModel.
+        // This should be moved to RepositoryManagementMainpageViewModel instead.
+        // Do that in the next PR.
+        var repos = QueryDatabaseForRepositories(removeRepositoriesNotInTheirSavedLocation);
+        return ConvertToLineItems(repos);
+    }
+
+    private IIncludableQueryable<Repository, RepositoryMetadata> QueryDatabaseForRepositories(bool removeRepositoriesNotInTheirSavedLocation)
+    {
+        var dbContext = _host.GetService<DevHomeDatabaseContext>();
+        var repositoriesFromDatabase = dbContext.Repositories.Include(x => x.RepositoryMetadata);
```

Comment:
Yes and yes.  Exceptions can be thrown on any kind of database access.

I have another branch with added try/catch support on database access.  It will have some more tests too.